src/Command/Notification/BulkActionCommand.php

```diff
@@ -0,0 +1,147 @@
+            ->addOption('environment', null, InputOption::VALUE_REQUIRED, 'EMS environment')
+            ->addOption('force', null, InputOption::VALUE_NONE, 'Do the bulk');
+    }
+
+    protected function initialize(InputInterface $input, OutputInterface $output): void
+    {
+        $this->io = new SymfonyStyle($input, $output);
+        $this->io->title('Bulk action notifications');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $actionId = \intval($input->getArgument('actionId'));
+        if (null === $action = $this->notificationService->getAction($actionId)) {
+            throw new \Exception(\sprintf('No notification action found with id %d', $actionId));
+        }
+
+        $rawQuery = \strval($input->getArgument('query'));
+        $query = \json_decode($rawQuery, true);
+        if (\json_last_error() > 0) {
+            throw new \RuntimeException(\sprintf('Invalid json query %s', $rawQuery));
+        }
+
+        $inputEnvironment = $input->getOption('environment');
+        $environmentName = $inputEnvironment ? \strval($inputEnvironment) : null;
+        if (null !== $environmentName) {
+            $environment = $this->environmentService->giveByName($environmentName);
+        } else {
+            $environment = $action->getContentType()->giveEnvironment();
+        }
+
+        $documents = $this->searchDocuments($environment, $query);
+
+        $this->io->block(\vsprintf('Found %s documents for notification action "%s" in %s', [
+            \count($documents),
+            $action->getName(),
+            $environment->getName(),
+        ]));
+
+        if (true !== $input->getOption('force')) {
+            $this->io->caution('For executing the bulk please rerun with --force');
+
+            return 0;
+        }
+
+        $username = \strval($input->getOption('username'));
+        $countSend = 0;
+        $progress = $this->io->createProgressBar(\count($documents));
+        $progress->start();
+
+        foreach ($documents as $document) {
+            if (null === $revision = $this->revisionService->getCurrentRevisionForDocument($document)) {
+                $this->io->warning(\sprintf('Could not find revision for ouuid "%s"', $document->getId()));
+                continue;
+            }
+
+            $added = $this->notificationService->addNotification($actionId, $revision, $environment, $username);
+
+            if ($added) {
+                ++$countSend;
+            }
+        }
+
+        $progress->finish();
+
+        $this->io->newLine(2);
+        $this->io->success(\sprintf('Created %d new notification with username "%s"', $countSend, $username));
+
+        return 0;
+    }
+
+    /**
+     * @param array<mixed> $query
+     *
+     * @return DocumentInterface[]
+     */
+    private function searchDocuments(Environment $environment, array $query): array
```

Comment:
Why nott returning an iterator? You load all document in memory

Reply:
Fixed used the count on the elasticSearch service